web-console/src/views/ingestion-view/ingestion-view.tsx

```diff
@@ -278,24 +283,37 @@ ORDER BY "rank" DESC, "created_time" DESC`;
     });
 
     this.taskQueryManager = new QueryManager({
-      processQuery: async capabilities => {
-        if (capabilities.hasSql()) {
-          return await queryDruidSql({
-            query: IngestionView.TASK_SQL,
+        processQuery: async capabilities => {
+          if (this.state.preferOverlord) {
+            if (capabilities.hasOverlordAccess()) {
+              const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
+              return IngestionView.parseTasks(resp.data);
+            } else if (capabilities.hasSql()) {
+              return await queryDruidSql({
+                query: IngestionView.TASK_SQL,
+              });
+            } else {
+              throw new Error(`must have SQL or overlord access`);
+            }
+          } else {
+            if (capabilities.hasSql()) {
+              return await queryDruidSql({
+                query: IngestionView.TASK_SQL,
+              });
+            } else if (capabilities.hasOverlordAccess()) {
+              const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
+              return IngestionView.parseTasks(resp.data);
+            } else {
+              throw new Error(`must have SQL or overlord access`);
+            }
+          }
+        },
```

Comment:
We could refactor this to remove the duplicate code and simply for reading.
Note: `fetchTasksViaOverlord` and `fetchTasksViaSql` are added as static functions below. Feel free to (re)name it as closer to druid platform terms.

```suggestion
    processQuery: async capabilities => {
        const { preferOverlord } = this.state;
        const hasOverlordAccess = capabilities.hasOverlordAccess();
        const hasSql = capabilities.hasSql();

        if (!hasOverlordAccess && !hasSql) {
          throw new Error(`must have SQL or overlord access`);
        }

        if (preferOverlord) {
          return hasOverlordAccess
            ? IngestionView.fetchTasksViaOverlord()
            : IngestionView.fetchTasksViaSql();
        } else {
          return hasSql ? IngestionView.fetchTasksViaSql() : IngestionView.fetchTasksViaOverlord();
        }
      },
```

Reply:
thanks @esakkiraj